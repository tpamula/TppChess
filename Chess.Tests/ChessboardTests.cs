using Chess.Core.Enums;
using Chess.Core.Model;
using System;
using System.Collections.Generic;
using Xunit;

namespace Chess.Tests
{
    public class ChessboardTests
    {
        [Fact]
        public void initial_setup_is_correct()
        {
            var chessboard = new Chessboard();

            foreach (int row in new[] { 1, 8 })
            {
                Assert.True(chessboard['a', row] == PieceType.Rook);
                Assert.True(chessboard['b', row] == PieceType.Knight);
                Assert.True(chessboard['c', row] == PieceType.Bishop);
                Assert.True(chessboard['d', row] == PieceType.Queen);
                Assert.True(chessboard['e', row] == PieceType.King);
                Assert.True(chessboard['f', row] == PieceType.Bishop);
                Assert.True(chessboard['g', row] == PieceType.Knight);
                Assert.True(chessboard['h', row] == PieceType.Rook);
            }

            foreach (int row in new[] { 2, 7 })
            {
                for (int i = 0; i < 8; i++)
                {
                    Assert.True(chessboard[Convert.ToChar(i + 97), row] == PieceType.Pawn);
                }
            }

            for (int i = 3; i <= 6; i++)
            {
                for (int j = 0; j < 8; j++)
                {
                    Assert.True(chessboard[Convert.ToChar(j + 97), i] == PieceType.Empty);
                }
            }
        }
    }
}