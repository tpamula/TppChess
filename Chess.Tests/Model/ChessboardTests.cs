using ApprovalTests;
using ApprovalTests.Reporters;
using Chess.Core.Enums;
using Chess.Core.Model;
using Xunit;

namespace Chess.Tests.Model
{
    [UseReporter(typeof(DiffReporter))]
    public class ChessboardTests
    {
        [Fact]
        public void initial_setup_is_correct()
        {
            var chessboard = new Chessboard();

            Approvals.Verify(chessboard.ToString());
        }

        [Fact]
        public void returns_correct_piece_on_a1_field()
        {
            var chessboard = new Chessboard();

            Assert.Equal(chessboard["a1"], new Piece(PieceColor.White, PieceType.Rook));
        }

        [Fact]
        public void returns_correct_piece_on_e8_field()
        {
            var chessboard = new Chessboard();

            Assert.Equal(chessboard["e8"], new Piece(PieceColor.Black, PieceType.King));
        }
    }
}