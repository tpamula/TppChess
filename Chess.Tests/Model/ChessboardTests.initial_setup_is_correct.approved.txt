BR BK BB BQ BK BB BK BR
BP BP BP BP BP BP BP BP
IE IE IE IE IE IE IE IE
IE IE IE IE IE IE IE IE
IE IE IE IE IE IE IE IE
IE IE IE IE IE IE IE IE
WP WP WP WP WP WP WP WP
WR WK WB WQ WK WB WK WR